pkg/istio-agent/health/health_check.go

```diff
@@ -69,9 +75,11 @@ func fillInDefaults(cfg *v1alpha3.ReadinessProbe) *v1alpha3.ReadinessProbe {
 		}
 		h.HttpGet.Scheme = strings.ToLower(h.HttpGet.Scheme)
 		if h.HttpGet.Host == "" {
-			// Kubernetes uses pod IP. However, the istio rewrite app probe uses localhost, so we
-			// should probably favor consistency with Istio than Kubernetes
-			h.HttpGet.Host = "localhost"
+			if legacyLocalhostProbeDestination.Get() {
+				h.HttpGet.Host = "localhost"
+			} else {
+				h.HttpGet.Host = "127.0.0.6"
```

Comment:
We don't want to send *to* 127.0.0.6, we want to have it be the local address. The reason is we have some special iptables rules that ensure this doesn't get captured.

I think we want to send to `proxy.IPAddresses[0]`, this is what use in the auto generate WorkloadEntry so if its wrong its at least consistent :slightly_smiling_face:

Reply:
How would I get `proxy.IPAddresses[0]` in the agent? Doesn't auto generation happen in pilot?

Reply:
its in main.go of pilot-agent. The agent sends that info to pilot so its shared

Reply:
The proxy sets that in the node metadata it sends to the control plane. That's initialized here in agent:

https://github.com/istio/istio/blob/20beaf5edcad3704d18dbf8d769b1658620939cc/pilot/cmd/pilot-agent/main.go#L97

Reply:
oh, i see. thanks guys